libdebug/libdebug.py

```diff
@@ -1373,15 +1414,15 @@ def debugger(
 
     debugging_context = create_context(debugger)
 
-    debugging_context.clear()
+    if not env:
```

Comment:
???

Reply:
It is the first time it has been instantiated, do we really need to clear it?
Moreover, it obviously creates problems with uninitialized variables like threads.

Reply:
In my original implementation the clear() method was called during initialization because that was the only method where certain attributes of the DebuggingContext were set, such as process_id=0 and pipe_manager=None.
If this is not the case anymore, then we can skip the call to clear.

Reply:
Actually this comment was referring to the `if not env: env = os.environ` branch

Reply:
Yup, I changed that behavior in this PR too :)

Reply:
The behavior is the same, re-adding these two lines which I removed last weekd just slows down the startup time at each `.run()` invocation.

See https://github.com/libdebug/libdebug/commit/cc67ff7a7e30d16e7c9a08635965d401aa5b1ced

Reply:
oh, I see your point. I was referring to the wrong lines. I will remove them then

Reply:
dd9ff8e30c757f979ef7738494f7a8ec3b59e284